source/lac/trilinos_vector_base.cc

```diff
@@ -372,7 +372,7 @@ namespace TrilinosWrappers
     // is zero on _all_ processors.
     const Epetra_MpiComm *mpi_comm
       = dynamic_cast<const Epetra_MpiComm *>(&vector->Map().Comm());
-    Assert(mpi_comm != 0, ExcInternalError());
+    AssertThrow(mpi_comm != 0, ExcInternalError());
```

Comment:
The point here was that we don't check whether the `dynamic_cast` fails in a Release build, if we only use `Assert`.